cmd/kubeadm/app/cmd/token_test.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"flag"
+	"regexp"
+	"testing"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	TokenExpectedRegex = "^\\S{6}\\.\\S{16}\n$"
+)
+
+var kubeadmPath string
+
+func init() {
+	flag.StringVar(&kubeadmPath, "kubeadm-path", "", "Location of kubeadm")
```

Comment:
What I'm saying is that we should move the tests that are executing a pre-built kubeadm binary into `cmd/kubeadm/test/integration` or a similar path (`cmd/kubeadm/app/integration`)

And in those tests we can execute the binary and verify that functionality is working properly.

In "normal" `gofile_test.go` files, we should have unit tests that call functions in `gofile.go`, just like @dgoodwin has done in https://github.com/kubernetes/kubernetes/pull/35632

@dmmcquay No we shouldn't centralize these integration test to `test/integration`, but have  kubeadm-only tests that are easy to take with us if/when we move to a separate repo

@pipejakob Do you know what I mean with this